### DISCUSSION TITLE
With the new App Router, are people putting 'use client' everywhere?

### DISCUSSION THREAD

**Discussion by elie222** [vercel/next.js#51471] [App Router] [answered]
I'm migrating an old app over, and I can see half the app just having `use client`. Or having use client at the top of every page?

Trying to understand the benefit of it right now. Even my home page which I would have thought could benefit from it a bit had to have it in most places.

**Comment by hesalx** [comment, marked as answer]
It depends on how exactly it breaks when you don't add `'use client'`.

It is easy to imagine that a lot of components will not work as server components right away if they were created without this choice. It was simply not possible to create some components in a different way previously.
For example, any time you rely on `useEffect` and/or `useState` for async data fetching, it could be just plain `async/await` in a server component. Or if you need cookies or headers, you can as well access them on the server from request headers instead of from `window`. 
Many third-party libraries still (and will for some time) rely implicitly on client-side or browser features for this reason.

[This page](https://nextjs.org/docs/getting-started/react-essentials#server-components) provides a nice diagram which shows that in general client-only components are "pushed" as close to the user as possible (highly interactive or browser-local features, `onClick`, `localStorage`, animations, etc.). The components are more granular and specialised (static vs data fetching vs interactive).

Unless you were aiming specifically at full SSR/SSG (for SEO or other reasons) it is less likely that the components in the project were designed and/or split in a way that makes them naturally fall into server vs client categories. 
If a reason to do so did not exist for you previously, I would argue that the App Router does not create such a reason on it's own. You should be good to go with a lot of `'use client'` if what you have haven't bothered you before the App Router, it just opens more possibilities for better code organisation and performance.
